src/prefect/tasks.py

```diff
@@ -210,6 +213,7 @@ def __init__(
         refresh_cache: Optional[bool] = None,
         on_completion: Optional[List[Callable[["Task", TaskRun, State], None]]] = None,
         on_failure: Optional[List[Callable[["Task", TaskRun, State], None]]] = None,
+        is_retriable: Optional[Callable[["Task", TaskRun, State], bool]] = None,
```

Comment:
Should this be shown to the user as an `on_retry` hook that can return `False` to not retry? An `is_retriable` callable option feels a little awkward.

Reply:
I was on the fence about that.

1. `on_retry` implies something other than "return true if a retry is okay. return false if not", so I went with `is_retriable`.
1. Documentation would have to note that this runs before any retry happens even if the retry gets rejected. Doesn't really map to the name `on_retry`.

Definitely open to better verbiage though.

Reply:
Perhaps a `retry_on_failure` parameter that can be a boolean or callable? Similar to how `task_run_name` can be a pre-defined string or generated on the spot. It would be true by default. How do you feel about overriding the retries parameters' role as the decider?

Reply:
I don't think you want a situation where someone could set `retry_on_failure=True` but not set `retries`. It's kind of awkward to duplicate those. You have a great point about `on_retry` feeling a bit wrong too though... hm. Perhaps `should_retry`? Curious to hear the thoughts of the Prefect team :)

Reply:
It may have been in another branch I started developing on, but I used the word 'ratify' somewhere. Maybe this is the term we're looking for?

Reply:
@dominictarro Thanks for coming up with a variety of possibilities! I have a few favorites, but I'll chat with the team today and update here before the EOD.

Reply:
Hi @dominictarro, we weren't able to come to a consensus today but we'll think on it more on Monday :)

Reply:
We settled on `retry_condition_fn` to match with `cache_key_fn` :)